src/main/kotlin/project2/Main.kt

```diff
@@ -71,13 +90,31 @@ fun safeResultsToCSV(results: List<Train>, outputFile: String = "results.csv") {
         i++
 
         if (result.getDelayed()){
-            row = arrayOf(i.toString(), result.getcurrentSegment().toString(), "delayed")
+            row = arrayOf(i.toString(), result.getcurrentSegment(0).toString(), "delayed")
         }
         else {
-            row = arrayOf(i.toString(), result.getcurrentSegment().toString(), "ontime")
+            row = arrayOf(i.toString(), result.getcurrentSegment(0).toString(), "ontime")
         }
         trainrows.add (row)
     }
 
     csvWriter.writeRowsAndClose(trainrows)
+}
+
+fun writeresultstoconsole(results: List<Train>) {
```

Comment:
Bitte verwende grundsätzlich Camel Case, d.h. beginn alle Wörter nach dem ersten mit einem Großbuchstaben (beim ersten hängt es davon ab, was benannt wird). D.h. in diesem Fall `writeResultsToConsole` - das macht den Code wesentlich lesbarer.